### PR TITLE
Use DefaultMetaGeneratorConfig in MetadataEnrichers to initialize configurations

### PR DESCRIPTION
**Pull request by ChrsMark**
Fixes https://github.com/elastic/beats/issues/13409#event-2594157033

### CODE REVIEW

File: libbeat/common/kubernetes/metadata.go [hunk 1]
```diff
@@ -52,6 +52,15 @@ type MetaGeneratorConfig struct {
...
 type metaGenerator = MetaGeneratorConfig
 
+// DefaultMetaGeneratorConfig initializes and returns a new MetaGeneratorConfig with default values
+func DefaultMetaGeneratorConfig() MetaGeneratorConfig {
+	return MetaGeneratorConfig{
+		IncludeCreatorMetadata: true,
+		LabelsDedot:            true,
+		AnnotationsDedot:       true,
+	}
+}
```
Comment (jsoriano) [inline libbeat/common/kubernetes/metadata.go]: For consistency could you also use this `DefaultMetaGeneratorConfig()` to initialize the `metaGenerator` in `NewMetaGenerator` some lines below?

File: libbeat/common/kubernetes/metadata.go [hunk 2]
```diff
@@ -52,6 +52,15 @@ type MetaGeneratorConfig struct {
```
Comment (jsoriano) [inline libbeat/common/kubernetes/metadata.go]: Not related, but could you reorder the fields in `MetaGeneratorConfig`? I think that the comment about undocummented fields only refers to `include_creator_metadata`.
```
type MetaGeneratorConfig struct {
	IncludeLabels      []string `config:"include_labels"`
	ExcludeLabels      []string `config:"exclude_labels"`
	IncludeAnnotations []string `config:"include_annotations"`

	LabelsDedot            bool `config:"labels.dedot"`
	AnnotationsDedot       bool `config:"annotations.dedot"`

	// Undocumented settings, to be deprecated in favor of `drop_fields` processor:
	IncludeCreatorMetadata bool `config:"include_creator_metadata"`
}
```

**Comment by ChrsMark** [comment]
@jsoriano thanks for reviewing!
Comments have been addressed and I also manually tested to confirm:
```yml
apiVersion: v1
kind: Pod
metadata:
  name: memory-demo
  namespace: kube-system
  labels:
    environment: production
    app: nginx
    app.kubernetes.io/name: botelastic
spec:
  containers:
  - name: detodedpod
    image: nginx
```
Screenshot 2019-08-30 at 10 41 12
Seems to be working as expected.

**Comment by ChrsMark** [comment]
> Code LGTM, let's wait to see why `container._module` appears in fields.

After rebasing on top of #13433 it works fine!
Screenshot 2019-08-30 at 16 09 58

**Comment by ChrsMark** [comment]
Failing Jenkins test seems irrelevant. I'm gonna merge this since Travis went green.

**Pull request merged**